src/Altinn.App.Core/Internal/Process/ExpressionsExclusiveGateway.cs

```diff
@@ -0,0 +1,168 @@
+            {
+                InstanceIdentifier instanceIdentifier = new InstanceIdentifier(instance);
+                var dataGuid = GetDataId(instance, checkedDataTypeId);
+                Type dataElementType = dataType;
+                if (dataGuid != null)
+                {
+                    data = await _dataClient.GetFormData(instanceIdentifier.InstanceGuid, dataElementType, instance.Org, instance.AppId.Split("/")[1], int.Parse(instance.InstanceOwner.PartyId), dataGuid.Value);
+                }
+            }
+
+            var state = await _layoutStateInit.Init(instance, data, layoutSetId: layoutSet?.Id, gatewayAction: action);
+            return state;
+        }
+
+        private bool EvaluateSequenceFlow(LayoutEvaluatorState state, SequenceFlow sequenceFlow)
+        {
+            if (sequenceFlow.ConditionExpression != null)
+            {
+                var expression = GetExpressionFromCondition(sequenceFlow.ConditionExpression);
+                foreach (var componentContext in state.GetComponentContexts())
+                {
+                    var result = ExpressionEvaluator.EvaluateExpression(state, expression, componentContext);
+                    if (result is bool boolResult && boolResult)
+                    {
+                        return true;
+                    }
+                }
+            }
+            else
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static Expression GetExpressionFromCondition(string condition)
+        {
+            JsonSerializerOptions options = new()
+            {
+                AllowTrailingCommas = true,
+                ReadCommentHandling = JsonCommentHandling.Skip,
+                PropertyNameCaseInsensitive = true,
+            };
+            Utf8JsonReader reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(condition));
+            reader.Read();
+            var expressionFromCondition = ExpressionConverter.ReadNotNull(ref reader, options);
+            return expressionFromCondition;
+        }
+
+        private LayoutSet? GetLayoutSet(Instance instance)
+        {
+            string taskId = instance.Process.CurrentTask.ElementId;
+            JsonSerializerOptions options = new() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+
+
+            string layoutSetsString = _resources.GetLayoutSets();
+            LayoutSet? layoutSet = null;
+            if (!string.IsNullOrEmpty(layoutSetsString))
+            {
+                LayoutSets? layoutSets = JsonSerializer.Deserialize<LayoutSets>(layoutSetsString, options)!;
+                layoutSet = layoutSets?.Sets?.FirstOrDefault(t => t.Tasks.Contains(taskId));
+            }
+
+            return layoutSet;
+        }
+
+        private async Task<(string? DataTypeId, Type? DataTypeClassType)> GetDataType(Instance instance, LayoutSet? layoutSet, string? dataTypeId)
```

Comment:
This logic seems to belong elsewhere, but we're perhaps missing the natural place to place it. Unless we use similar methods in other parts of the code we should make a note of it and figure out where it should be until next time.

Reply:
Couldn't find a natural home, I will add a TODO comment stating that we should find a home for it